postings.go

```diff
@@ -158,24 +154,31 @@ func Merge(its ...Postings) Postings {
 }
 
 type mergePostings struct {
-	a, b     Postings
-	aok, bok bool
-	cur      uint32
+	a, b        Postings
+	initialized bool
+	aok, bok    bool
+	cur         uint32
 }
 
 func newMergePostings(a, b Postings) *mergePostings {
-	it := &mergePostings{a: a, b: b}
-	it.aok = it.a.Next()
-	it.bok = it.b.Next()
-
-	return it
+	return &mergePostings{a: a, b: b, aok: true, bok: true}
 }
 
 func (it *mergePostings) At() uint32 {
 	return it.cur
 }
 
 func (it *mergePostings) Next() bool {
+	if !it.initialized {
+		if !it.a.Next() {
+			it.aok = false
+		}
+		if !it.b.Next() {
+			it.bok = false
+		}
+		it.initialized = true
+	}
```

Comment:
Oh I see, didn't see the bigger picture including the cost.
What would the cost function be based on? Number of elements or anything further as well?

Given relevant optimisations we can of course store additional meta data as well.